pkg/configuration/environment.go

```diff
@@ -0,0 +1,122 @@
+package configuration
+
+import (
+	"context"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/subosito/gotenv"
+)
+
+// LoadConfiguredEnvironment updates the environment with local configuration. Precedence as follows:
+//  1. std folder-based config files
+//  2. given command-line parameter config file
+//  3. std config file in home directory
+//  4. global shell configuration
+func LoadConfiguredEnvironment(config Configuration) {
+	parsedEnv := getParsedEnvFromShell("bash")
+	shell := parsedEnv["SHELL"]
+	fromSpecificShell := getParsedEnvFromShell(shell)
+
+	if len(fromSpecificShell) > 0 {
+		SetParsedVariablesToEnv(fromSpecificShell, false)
+	} else {
+		SetParsedVariablesToEnv(parsedEnv, false)
+	}
+
+	for _, file := range config.GetStringSlice(CUSTOM_CONFIG_FILES) {
+		if !filepath.IsAbs(file) {
+			file = filepath.Join(config.GetString(WORKING_DIRECTORY), file)
+		}
+		loadFile(file)
+	}
+}
+
+func loadFile(fileName string) {
+	file, err := os.Open(fileName)
+	if err != nil {
+		return
+	}
+	defer func(file *os.File) { _ = file.Close() }(file)
+	env := gotenv.Parse(file)
+
+	// we want these settings to overwrite existing settings (apart from the path)
+	SetParsedVariablesToEnv(env, true)
+}
+
+func SetParsedVariablesToEnv(env gotenv.Env, replaceVars bool) {
+	for k, v := range env {
+		if k == "PATH" {
+			UpdatePath(v)
+			continue
+		}
+
+		_, exists := os.LookupEnv(k)
+		if !exists || replaceVars {
+			_ = os.Setenv(k, v)
+		}
+	}
+}
+
+func getParsedEnvFromShell(shell string) gotenv.Env {
+	// guard against command injection
+	var shellWhiteList = map[string]bool{
+		"bash":      true,
+		"/bin/zsh":  true,
+		"/bin/sh":   true,
+		"/bin/fish": true,
+		"/bin/csh":  true,
+		"/bin/ksh":  true,
+		"/bin/bash": true,
+	}
+
+	if !shellWhiteList[shell] {
+		return gotenv.Env{}
+	}
+
+	ctx, cancelFunc := context.WithTimeout(context.Background(), 2*time.Second)
+	defer cancelFunc()
+
+	// deepcode ignore CommandInjection: false positive
+	env, err := exec.CommandContext(ctx, shell, "--login", "-i", "-c", "printenv && exit").Output()
```

Comment:
Question: I'm struggling to understand how this works in a cross platform environment. Especially the windows case seems to be missing.

Reply:
I accidentally deleted the windows early return. Oops.